Jenkinsfile

```diff
@@ -19,63 +25,114 @@ pipeline {
-                            ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION -v"""
-                        archiveArtifacts artifacts: "doc/_build/**", allowEmptyArchive: true
+                        dir("${REPO}") {
+                            warnError("Docs") {
+                                checkout scm
+                                sh "docker pull ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION"
+                                sh """docker run -u "\$(id -u):\$(id -g)" \
+                                      --rm \
+                                      -v \$(pwd):/build \
+                                      ghcr.io/xmos/xmosdoc:$XMOSDOC_VERSION -v html latex"""
+
+                                // Zip and archive doc files
+                                zip dir: "doc/_build/html", zipFile: "${REPO}_docs_html.zip"
+                                archiveArtifacts artifacts: "${REPO}_docs_html.zip"
+                                archiveArtifacts artifacts: "doc/_build/pdf/${REPO}*.pdf"
+                            }
+                        }
                     }
                     post {
                         cleanup {
                             xcoreCleanSandbox()
                         }
                     }
                 }
-                stage('Basic tests') {
-                    when {
-                        expression { !env.GH_LABEL_DOC_ONLY.toBoolean() }
+                stage('XCommon build') {
+                    agent {
+                        label "x86_64 && linux"
+                    }
+                    steps {
+                        dir("${REPO}") {
+                            checkout scm
+                            dir("tests") {
+                                withTools(params.TOOLS_VERSION) {
+                                    sh 'cmake -B build -G "Unix Makefiles"'
+                                    // Note no -B build so builds the xcommon Makefile
+                                    sh "xmake all -j 16"
+                                }
+                                archiveArtifacts artifacts: "**/*.xe", allowEmptyArchive: true
+                            }
+                        }
                     }
+                    post {
+                        cleanup {
+                            xcoreCleanSandbox()
+                        }
+                    }
+                }
+                stage('Custom CMake build') {
+                    agent {
+                        label "x86_64 && linux" 
+                    }
+                    steps {
+                        dir("${REPO}") {
+                            checkout scm
+                            sh 'git submodule update --init --recursive --depth 1'
+                            sh "wget https://raw.githubusercontent.com/xmos/xmos_cmake_toolchain/main/xs3a.cmake"
+                            sh "wget https://raw.githubusercontent.com/xmos/xmos_cmake_toolchain/main/xc_override.cmake"
```

Comment:
Since you are still running the tests on the custom cmake build, I think it would be nicer to clone xmos_cmake_toolchain into the repo like we would for test dependencies like xtagctl, etc. rather than grabbing the required files like this.

Reply:
Yep fair enough - this was likely an optimisation by the original author. Now addressed with checkout and tag (V1.0.0)